src/impl_/pymethods.rs

```diff
@@ -288,3 +320,16 @@ where
         self.map(|o| o.into_py(py))
     }
 }
+
+fn extract_c_string(src: &'static str, err_msg: &'static str) -> PyResult<Cow<'static, CStr>> {
+    let bytes = src.as_bytes();
+    if bytes.last() == Some(&0) {
+        // Last byte is a nul; try to create as a CStr
+        let c_str = CStr::from_bytes_with_nul(bytes).map_err(|_| PyValueError::new_err(err_msg))?;
+        Ok(Cow::Borrowed(c_str))
+    } else {
+        // Allocate a new CString for this
+        let c_string = CString::new(bytes).map_err(|_| PyValueError::new_err(err_msg))?;
+        Ok(Cow::Owned(c_string))
+    }
+}
```

Comment:
Can we handle a `""` by pointing to a static `"\0"` instead? 

e.g. 
```
match bytes {
        [] => point to static "\0"
        [.., 0] => try &'static CStr
        _ => try CString
    }
```

Reply:
Oooh, I like this, will try now!